recipe/meta.yaml

```
@@ -81,63 +83,60 @@ outputs:
       dev_url: https://github.com/NVIDIA/cuda-python
 
   - name: cuda-bindings
-    version: {{ bindings_ver }}
+    version: {{ version }}
+    # Keep scripts under outputs section to ensure building the modules/packages provided by cuda-python
     script: build_output.sh   # [linux]
     script: build_output.bat  # [win]
     build:
       number: {{ number }}
-      # cuda.bindings needs to be built with the matching major.minor headers
-      {% if not (environ.get("cuda_compiler_version")|string()) == (bindings_major_version ~ "." ~ bindings_minor_version) %}
-      skip: true
-      {% else %}
-      skip: true  # [not build_bindings]
-      {% endif %}
+      skip: {{not build_bindings}}
       script_env:
         # Ensure that CUDA includes can be found by the host compiler
         - CUDA_HOME={{ PREFIX ~ subdir ~ os.pathsep ~ BUILD_PREFIX ~ subdir }}
       run_exports:
-        - {{ pin_subpackage('cuda-python', min_pin='x', max_pin='x') }}
+        - {{ pin_subpackage('cuda-bindings', min_pin='x.x', max_pin='x.x') }}
       ignore_run_exports_from:
         - {{ compiler('cuda') }}
         # The run-exports from these dev packages are too tight, so we ignore them.
         # The cuda-python package supports CUDA Enhanced Compatibility through its
         # use of dlopen, so it is compatible with other CUDA minor versions.
         - cuda-cudart-dev
         - cuda-nvrtc-dev
+      ignore_prefix_files:
+        - '**/*.c'
+        - '**/*.cpp'
+        - '**/*.so.c'
     requirements:
       build:
+        - {{ stdlib('c') }}
         - {{ compiler('c') }}
         - {{ compiler('cxx') }}
         - {{ compiler('cuda') }}
-        - {{ stdlib('c') }}
-        - cross-python_{{ target_platform }}  # [build_platform != target_platform]
-        - python                              # [build_platform != target_platform]
-        - cython                              # [build_platform != target_platform]
-        - cuda-version {{ bindings_major_version }}.{{ bindings_minor_version }}
+        - {{ pin_compatible('cuda-version', min_pin='x.x', max_pin='x.x') }}
       host:
-        - cuda-version {{ bindings_major_version }}.{{ bindings_minor_version }}
-        - cuda-cudart-dev
-        - cuda-cudart
-        - cuda-nvrtc-dev
-        - cuda-nvrtc
-        - cuda-profiler-api
-        - cython
-        - pip
-        - pyclibrary
         - python
+        - pip
         - setuptools
+        - cython
+        - pyclibrary 0.3.0
+        - {{ pin_compatible('cuda-version', min_pin='x.x', max_pin='x.x') }}
+        - {{ pin_compatible('cuda-cudart-dev', min_pin='x.x', max_pin='x.x') }}
+        - {{ pin_compatible('cuda-cudart', min_pin='x.x', max_pin='x.x') }}
+        - {{ pin_compatible('cuda-nvrtc-dev', min_pin='x.x', max_pin='x.x') }}
+        - {{ pin_compatible('cuda-nvrtc', min_pin='x.x', max_pin='x.x') }}
+        - {{ pin_compatible('cuda-profiler-api', min_pin='x.x', max_pin='x.x') }}
```

```suggestion
        - cuda-version 12.8
        - cuda-cudart-dev 12.8
        - cuda-cudart 12.8
        - cuda-nvrtc-dev 12.8
        - cuda-nvrtc 12.8
        - cuda-profiler-api 12.8
```
pin_compatible result in no pinning here. Instead you'd want to pin to a specific version.

(You may have to adjust the pins)